clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/HonoKafkaConsumer.java

```diff
@@ -69,19 +72,23 @@
 
     protected final Logger log = LoggerFactory.getLogger(getClass());
     protected final Vertx vertx;
-    protected final AtomicBoolean stopped = new AtomicBoolean();
+    protected final AtomicBoolean stopCalled = new AtomicBoolean();
```

Comment:
At the point in time when `stopCalled` is set to `true` (on the vert.x event loop thread), the consumer isn't stopped yet, just the `stop()` method has been called.
This distinction is relevant now because the variable is also checked from the Kafka polling thread, meaning it may be read between `stop()` getting invoked and the inner invocation of `consumer.close()` on the Kafka polling thread.